pip/pip-320.md

```diff
@@ -0,0 +1,241 @@
+documentation about it. In this very short paragraph I'll describe what you need to know about OTel from this PIP
+perspective.
+
+OpenTelemetry is a project aimed to standardize the way we instrument, collect and ship metrics from applications
+to telemetry backends, be it databases (e.g. Prometheus, Cortex, Thanos) or vendors (e.g. Datadog, Logz.io).
+It is divided into API, SDK and Collector:
+- API: interfaces to use to instrument: define a counter, record values to a histogram, etc.
+- SDK: a library, available in many languages, implementing the API, and other important features such as
+reading the metrics and exporting it out to a telemetry backend or OTel Collector. 
+- Collector: a lightweight process (application) which can receive or retrieve telemetry, transform it (e.g.
+filter, drop, aggregate)  and export it (e.g. send it to various backends). The SDK supports out-of-the-box 
+exporting metrics as Prometheus HTTP endpoint or sending them out using OTLP protocol. Many times companies choose to
+ship to the Collector and there ship to their preferred vendors, since each vendor already published their exporter
+plugin to OTel Collector. This makes the SDK exporters very light-weight as they don't need to support any 
+vendor. It's also easier for the DevOps team as they can make OTel Collector their responsibility, and have
+application developers only focus on shipping metrics to that collector.
+
+Just to have some context: Pulsar codebase will use the OTel API to create counters / histograms and records values to 
+them. So will the Pulsar plugins and Pulsar Function authors. Pulsar itself will be the one creating the SDK
+and using that to hand over an implementation of the API where ever needed in Pulsar. Collector is up to the choice
+of the user, as OTel provides a way to expose the metrics as `/metrics` endpoint on a configured port, so Prometheus
+compatible scrapers can grab it from it directly. They can also send it via OTLP to OTel collector.
+
+## Telemetry layers
+PIP-264 clearly outlined there will be two layers of metrics, collected and exported, side by side: OpenTelemetry 
+and the existing metric system - currently exporting in Prometheus. This PIP will explain in detail how it will work. 
+The basic premise is that you will be able to enable or disable OTel metrics, alongside the existing Prometheus 
+metric exporting.
+
+## Why OTel in Pulsar will be marked experimental and not GA
+As specified in [PIP-264]((pip-264.md), OpenTelemetry Java SDK has several fixes the Pulsar community must 
+complete before it can be used in production. They are [documented](pip-264.md#what-we-need-to-fix-in-opentelemetry)
+in PIP-264. The most important one is reducing memory allocations to be negligible. OTel SDK is built upon immutability, 
+hence allocated memory in O(`#topics`) which is a performance killer for low latency application like Pulsar. 
```

Comment:
Yes. 
I'm currently working on my 4th PR to OTel on the memory allocation decreasing. Some of it is already there, but not all of it.
It can not be GA without the final version of OTel with all my fixes.